pyfar/utils.py

```diff
@@ -127,3 +127,92 @@ def broadcast_cdims(signals, cdim=None):
     if cdim is None:
         cdim = np.max([len(s.cshape) for s in signals])
     return [broadcast_cdim(s, cdim) for s in signals]
+
+
+def concatenate(signals, caxis=0, broadcasting=False, comment=""):
+    """
+    Merge multiple Signal objects along a given axis. The signals are copied,
+    if needed broadcasted in largest dimenson and to a common cshape and a new
+    object is returned.
+
+    The :py:mod:`cshape <pyfar._concepts.audio_classes>` of the signals are
+    broadcasted following the `numpy broadcasting rules
+    <https://numpy.org/doc/stable/user/basics.broadcasting.html>`_
+
+    Parameters
+    ----------
+    signals : tuple of Signal
+        The signals to concatenate. Must either have the same cshape or be
+        broadcastable to the same cshape, except in the dimension corresponding
+        to axis (the first, by default). If this is the case, set
+        ``broadcasting=True``.
+    caxis : int
+        The axis along which the signals are concatenated.
+        The default is ``0``.
+    broadcasting: bool
+        If this is ``True``, the signals will be broadcasted into largest
+        dimension and into a common cshape, except of the caxis along which the
+        signals are concatenated. The default is ``False``.
+    comment: string
+        A comment related to the merged `data`. The default is ``""``, which
+        initializes an empty string. Comments of the input signals will also
+        be returned in the new Signal object. These comments are marked with
+        their corresponding signal position in the input tuple.
+    Returns
+    -------
+    merged : Signal
+        The merged signal object.
+    """
+    # check input
+    for signal in signals:
+        if not isinstance(signal, pf.Signal):
+            raise TypeError("All input data must be of type pf.Signal")
+    if not isinstance(comment, str):
+        raise TypeError("'comment' needs to be a string.")
+    if not isinstance(broadcasting, bool):
+        raise TypeError("'broadcasting' needs to be False or True.")
+    # check matching meta data of input signals.
+    [signals[0]._assert_matching_meta_data(s) for s in signals]
+    # broadcast signals into largest dimension and common cshapes
+    if broadcasting is True:
+        # broadcast signals into common cshape
+        cshapes = [s.cshape for s in signals]
+        max_cdim = np.max([len(sh) for sh in cshapes])
+        for i, sh in enumerate(cshapes):
+            for _ in range(max_cdim-len(sh)):
+                cshapes[i] = (1,) + cshapes[i]
+        # Finds broadcast cshape without the caxis to ignore
+        cshape = np.broadcast_shapes(*np.delete(cshapes, caxis,
+                                     axis=-1))
+        broad_signals = []
+        for i, s in enumerate(signals):
```

Comment:
using
```suggestion
        for signal, cshape in zip(signals, cshapes):
```
might be better to read and avoid indexing